src/main/java/org/spongepowered/api/event/server/QueryEvent.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.server;
+
+import org.spongepowered.api.event.GameEvent;
+import org.spongepowered.api.text.message.Message;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Called when the server is queried through the Query protocol.
+ */
+public interface QueryEvent extends GameEvent {
+
+    /**
+     * Gets the MOTD to respond with.
+     *
+     * <p>By default, this is the server's current MOTD</p>
+     *
+     * @return The MOTD to respond with.
+     */
+    String getMotd();
+
+    /**
+     * Sets the MOTD to respond with.
+     *
+     * @param motd The MOTD to respond with
+     */
+    void setMotd(String motd);
+
+    /**
+     * Gets the GameType to respond with.
+     *
+     * <p>By default, this is <code>SMP</code>.</p>
+     *
+     * @return The GameType to respond with
+     */
+    String getGameType();
+
+    /**
+     * Sets the GameType to respond with.
+     *
+     * @param gameType The GameType to respond with
+     */
+    void setGameType(String gameType);
+
+    /**
+     * Gets the map (world) to respond with.
+     *
+     * <p>By default, this is the current world on the server.</p>
+     *
+     * @return The map to respond with
+     */
+    String getMap();
```

Comment:
Maybe this would be better named getWorldName() ?

I'd wait for more of the team to comment before changing it however.